### PR TITLE
Support simple token-based form authentication

### LINKED ISSUE #24: Support simple token-based form authentication
As the initial step to support k8s for hawtio-online hawtio/hawtio-online#23, implement a simple token-based form authentication similar to one used in the k8s dashboard. See:
https://github.com/kubernetes/dashboard/blob/master/docs/user/access-control/creating-sample-user.md

### PR DESCRIPTION
**Pull request by tadayosi**
Fix  #24

### CODE REVIEW

**Comment by tadayosi** [comment]
@astefanutti Could you have a look when you have time?

**Review by oscerd — APPROVED** [review]

**Review by astefanutti — APPROVED** [review]
Awesome! The login page looks gorgeous 👍🏼!

**Comment by tadayosi** [comment]
@astefanutti @oscerd Thanks! Let's merge it.

File: plugins/form/login/login.component.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          const masterUri: string = _.get(window, 'HAWTIO_OAUTH_CONFIG.master_uri');
+          if (!masterUri) {
+            log.error('Master URI is not found');
+            this.invalidToken = true;
+            this.errorMessage = "Cannot validate token";
+            return;
+          }
+
+          this.$http
+            .get(masterUri, {
```
Comment (astefanutti) [inline plugins/form/login/login.component.ts:54]: Just a late comment, while doing some testing, I realise some service accounts may not be granted permission to access the root path of the API server, while there are generally granted permission to access the `api` path. This somehow relates to https://kubernetes.io/docs/tasks/administer-cluster/access-cluster-api/#without-kubectl-proxy.

Comment (tadayosi) [reply plugins/form/login/login.component.ts:54]: Thanks for testing, good catch!  I'll create another PR to fix it by using the master `/api` path.

**Pull request merged**